File: c++/2231.cpp
// My answer<way1>
// #include <iostream>
// #include <string>

// using namespace std;

// int main(){
//   int n, result;
//   cin >> n;

//   string s = to_string(n);
//   int size = s.size();
  
//   for(int i = n-9*size; i <= n-size; i++){
//     result = 0;
//     s = to_string(i);
//     for(int j=0; j<s.size(); j++){
//       result += (s[j]-'0');
//     }
//     result += i;
//     if(result == n){
//       cout << i;
//       return 0;
//     }
//   }
//   cout << 0;
  
//   return 0;
// }

// another answer<way2> 
#include <iostream>

using namespace std;

int soulution(int a){
  int sum = a;
  while(a){
    sum += (a%10);
    a /= 10;
  }

  return sum;
}

int main(){
  int n, sum;
  cin >> n;
  
  for(int i = 0; i <= n; i++){
    sum = soulution(i);
    if(sum == n){
      cout << i;
      return 0;
    }
  }
  cout << 0;
  
  return 0;
}

File: js/2231.js
const fs = require('fs');

const input = fs.readFileSync('/dev/stdin').toString();
const n = input.length;
const numberInput = Number(input);
const start = numberInput - n * 9;

let result = 0;

for (let i = start; i < numberInput; i++) {
  let sum = i;
  const stringValue = i.toString();
  for (let j = 0; j < stringValue.length; j++) {
    sum += Number(stringValue[j]);
  }

  if (sum === numberInput) {
    result = i;
    break;
  }
}

console.log(result);
